galasa-extensions-parent/dev.galasa.cps.etcd/src/main/java/dev/galasa/cps/etcd/internal/Etcd3CredentialsStore.java

```diff
@@ -136,10 +135,80 @@ public void setCredentials(String credentialsId, ICredentials credentials) throw
     @Override
     public void deleteCredentials(String credentialsId) throws CredentialsException {
         try {
-            deletePrefix("secure.credentials." + credentialsId);
+            deletePrefix(CREDS_PROPERTY_PREFIX + "." + credentialsId);
         } catch (InterruptedException | ExecutionException e) {
             Thread.currentThread().interrupt();
             throw new CredentialsException("Failed to delete credentials", e);
         }
     }
+
+    @Override
+    public Map<String, ICredentials> getAllCredentials() throws CredentialsException {
```

Comment:
getting all the credentials seems a very scary thing to be doing no ? Why do we need to do this ?

Reply:
It was a requirement documented in the story - https://github.com/galasa-dev/projectmanagement/issues/1467

This is used by the `GET /secrets` endpoint, which is used in `galasactl secrets get` - should this only return the keys and metadata?